src/renderer/vt/paint.cpp

```diff
@@ -379,13 +382,18 @@ using namespace Microsoft::Console::Types;
     RETURN_IF_FAILED(_MoveCursor(coord));
 
     std::wstring unclusteredString;
-    unclusteredString.reserve(clusters.size());
     short totalWidth = 0;
-    for (const auto& cluster : clusters)
+    unclusteredString.reserve(40);

```

Comment:
I'm not a fan of magic number. But without reserving space, this will run much slower. Should we pass size alongside `clusterIter`? I'm not sure about this.

Reply:
Uh... perhaps preallocate the number of cells width from `coord` to the right edge of the known dirty rect (see impl of `GetDirtyRectInChars`)

All of the engines should know a little bit about what area is actually invalid and they're always given a drawing coordinate.

Failing this, instead of allocating and reserving on each call to `_Paint*BufferLine`, they might be able to allocate the buffer when the `StartPaint()` comes in as a dirty-rect-wide string/vector and just re-use that one for every row and free it when the `EndPaint()` comes around. That'd make way fewer allocations per frame than one per call.